docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,763 @@
+
+### Changing the currently active theme
+
+`registerThemeInjector()` will return a handle to the `themeInjector` that can be used to change the active theme. Applications can call `themeInjector.set()`, passing in a new theme object, which will invalidate all themed widgets in the application tree and re-render them using the new theme.
+
+To simplify the process of changing themes and to allow for easier dynamic switching in a running application, a `ThemeSwitcher` utility widget is also available from `@dojo/framework/widget-core/mixins/Themed`.
+
+#### `ThemeSwitcher` Properties
+
+-   `renderer: (updateTheme(theme: Theme) => void): DNode | DNode[]`
+    -   The `renderer` that is called with an `updateTheme` function that can be used to switch themes based on user selection, for example. The `renderer` implementation should return `DNode | DNode[]`, which are the elements that will be rendered to allow theme selection within the application.
+-   `registryLabel ?: string`
+    -   (Optional) The registry label used to register the theme injector. When using the `registerThemeInjector` utility function, this property does not need to be set.
+
+#### `ThemeSwitcher` Example Usage
+
+The following example shows a themeable widget that uses `ThemeSwitcher` to render two buttons that allow users to switch between a `light` and `dark` theme.
+
+> src/widgets/MyApp.ts
+
+```ts
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import ThemedMixin, { ThemeSwitcher, theme, UpdateTheme } from '@dojo/framework/widget-core/mixins/Themed';
+
+import lightTheme from '../themes/light-theme/theme';
+import darkTheme from '../themes/dark-theme/theme';
+import * as css from './styles/MyApp.m.css';
+
+@theme(css)
+class MyApp extends ThemedMixin(WidgetBase) {
+	protected render() {
+		return v('div', [
+			w(ThemeSwitcher, {
+				renderer: (updateTheme: UpdateTheme) => {
+					return v('div', [
+						v(
+							'button',
+							{
+								onclick: () => {
+									updateTheme(lightTheme);
+								}
+							},
+							['light']
+						),
+						v(
+							'button',
+							{
+								onclick: () => {
+									updateTheme(darkTheme);
+								}
+							},
+							['dark']
+						)
+					]);
+				}
+			}),
+			v('div', { classes: this.theme(css.container) })
+		]);
+	}
+}
+```
+
+## Making themeable widgets
+
+There are three requirements for widgets to be considered themeable:
+
+1.  The widget's class should have the `@theme()` decorator applied, passing in the widget's imported CSS module as a decorator argument.
+    -   Note that widgets can import and reference multiple CSS modules, in which case additional `@theme()` decorators can be stacked, one for each corresponding CSS import intended to be themeable.
+2.  The widget's parent class should be wrapped with `ThemedMixin`.
```

Comment:
Is this accurate? Is it not simply that the widget's class should extend `ThemedMixin`?